ui/components/app/signature-request/signature-request-message/signature-request-message.js

```diff
@@ -0,0 +1,120 @@
+import React, { useContext, useState } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { debounce } from 'lodash';
+import classnames from 'classnames';
+import { I18nContext } from '../../../../contexts/i18n';
+import { getMetaMaskIdentities, getAccountName } from '../../../../selectors';
+import Address from '../../transaction-decoding/components/decoding/address';
+import {
+  isValidHexAddress,
+  toChecksumHexAddress,
+} from '../../../../../shared/modules/hexstring-utils';
+
+export default function SignatureRequestMessage({
+  data,
+  onMessageScrolled,
+  setMessageRootRef,
+  messageRootRef,
+  messageIsScrollable,
+}) {
+  const t = useContext(I18nContext);
+  const [messageIsScrolled, setMessageIsScrolled] = useState(false);
+  const identities = useSelector(getMetaMaskIdentities);
+  const setMessageIsScrolledAtBottom = () => {
+    if (!messageRootRef || messageIsScrolled) {
+      return;
+    }
+
+    const { scrollTop, offsetHeight, scrollHeight } = messageRootRef;
+    const isAtBottom = Math.round(scrollTop) + offsetHeight >= scrollHeight;
+
+    if (isAtBottom) {
+      setMessageIsScrolled(true);
+      onMessageScrolled();
+    }
+  };
+
+  const renderNode = (renderData) => {
+    return (
+      <div className="signature-request-message--node">
```

Comment:
So, a few things.

1. This isn't proper BEM syntax for class names `--` is a modifier spacer and `__` is a block spacer. so i think we'd want `signature-request-message__node`.
2. Later you have `--node-label` and use `&-label` to style it, this also isn't proper BEM syntax, node-label would be its own new block not a nested block.
3. I realize you are trying to use the existing styles here, but I would prefer that if you are going to remove old patterns (thank you!!! this is awesome!!!) by refactoring class components to functional, dropping the '.component' syntax that we also use `Box` and `Typography` to ease in the transition to the new design system in the future.

let me know if that makes sense

Reply:
You are right @brad-decker, sorry for that! I know that we are using `Box` and `Typography` by refactoring class components to functional and I know about proper BEM syntax, sorry for that once again! I will take care of this in the future PRs! 
I left only one `div` because it has the `ref` property. I have not been able to do this with the `Box` component, because `Box` component don't have that property. If it is okay to stay like this great, but if you have some suggestions about this please comment and I will make changes. I made changes [here](https://github.com/MetaMask/metamask-extension/pull/16018/commits/d3277889201306fd0a3cb313ea0d6b2f59181a0b).
Thank you in advance!